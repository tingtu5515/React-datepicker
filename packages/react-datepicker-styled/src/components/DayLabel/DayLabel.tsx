import React from 'react'
import Text from '../Text'

interface MonthLabelProps {
  label: string
}

const MonthLabel = ({label}: MonthLabelProps) => {
  return (
    <Text fontFamily="Montserrat" fontSize="11px" fontWeight={500} color="#929598">
      {label}
    </Text>
  )
}

export default MonthLabel
