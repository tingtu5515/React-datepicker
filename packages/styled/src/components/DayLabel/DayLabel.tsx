import React, {useContext} from 'react'
import {ThemeContext} from 'styled-components'
import Text from '../Text'
// eslint-disable-next-line import/no-unresolved
import {DayLabelTheme} from '../../@types/theme'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'
import getThemeProp from '../../utils/getThemeProp'

interface MonthLabelProps {
  label: string
}

const MonthLabel = ({label}: MonthLabelProps) => {
  const themeContext = useContext(ThemeContext)

  const theme: DayLabelTheme = useThemeProps({
    fontFamily: globalStyles.fontFamily,
    dayLabelColor: getThemeProp('silverCloud', globalStyles.colors.silverCloud, themeContext),
    dayLabelFontWeight: 500,
    dayLabelFontSize: '11px',
  })
  return (
    <Text
      fontFamily={theme.fontFamily}
      fontSize={theme.dayLabelFontSize}
      fontWeight={theme.dayLabelFontWeight}
      // @ts-ignore
      color={theme.dayLabelColor}
      data-testid="DayLabel"
    >
      {label}
    </Text>
  )
}

export default MonthLabel
