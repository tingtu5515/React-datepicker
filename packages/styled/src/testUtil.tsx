import React from 'react'
import {render} from '@testing-library/react'
import DatepickerContext, {datepickerContextDefaultValue} from './context/datepickerContext'

function renderWithProviders(ui: any, options?: any, contextValues?: Record<string, unknown>) {
  return render(
    <DatepickerContext.Provider value={{...datepickerContextDefaultValue, ...contextValues}}>
      {ui}
    </DatepickerContext.Provider>,
    options,
  )
}

export * from '@testing-library/react'
export {renderWithProviders as render}
