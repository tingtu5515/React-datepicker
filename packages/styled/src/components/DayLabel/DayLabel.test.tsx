import * as React from 'react'
import {render} from '../../testUtil'
import DayLabel from '.'

test('should match snapshot', () => {
  const {container} = render(<DayLabel label="Test" />)
  expect(container).toMatchSnapshot()
})
