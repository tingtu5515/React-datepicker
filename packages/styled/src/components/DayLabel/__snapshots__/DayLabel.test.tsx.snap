// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
  color: #929598;
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
}

<div>
  <div
    class="c0"
    color="#929598"
    data-testid="DayLabel"
    font-family="Montserrat, sans-serif"
    font-size="11px"
    font-weight="500"
  >
    Test
  </div>
</div>
`;
